duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -256,21 +257,35 @@ public void useCustomCertificates(String[] customCaCerts) {
     httpClient = httpClient.newBuilder().certificatePinner(pinner).build();
   }
 
-  protected String canonRequest(String date, int sigVersion)
+  protected String canonRequest(String date, int inSigVersion)
       throws UnsupportedEncodingException {
+
+    int[] avaiableSigVersion = {1, 2};
+    int sigVersion = 1;
+
+    if (Arrays.stream(avaiableSigVersion).anyMatch(i -> i == inSigVersion)){
+      sigVersion = inSigVersion;
+    }
+
     String canon = "";
-    if (sigVersion == 2) {
-      canon += date + "\n";
+    if (sigVersion == 1) {
+      canon += method.toUpperCase() + System.lineSeparator();
+      canon += host.toLowerCase() + System.lineSeparator();
+      canon += uri + System.lineSeparator();
+      canon += canonQueryString();
+    }
+    else if (sigVersion == 2) {
+      canon += date + System.lineSeparator();
+      canon += method.toUpperCase() + System.lineSeparator();
+      canon += host.toLowerCase() + System.lineSeparator();
+      canon += uri + System.lineSeparator();
+      canon += canonQueryString();
     }
-    canon += method.toUpperCase() + "\n";
-    canon += host.toLowerCase() + "\n";
-    canon += uri + "\n";
-    canon += createQueryString();
 
     return canon;
   }
 
-  private String createQueryString()
+  private String canonQueryString()
```

Comment:
Rename it `canon` to match feature `caonX` related methods.

Reply:
good call